control-plane/pkg/reconciler/base/reconciler.go

```diff
@@ -0,0 +1,160 @@
+	BrokersTriggersConfigMapName      string
+	Format                            string
+	SystemNamespace                   string
+}
+
+func (r *Reconciler) GetBrokersTriggersConfigMap() (*corev1.ConfigMap, error) {
+	return r.KubeClient.CoreV1().
+		ConfigMaps(r.BrokersTriggersConfigMapNamespace).
+		Get(r.BrokersTriggersConfigMapName, metav1.GetOptions{})
+}
+
+// getBrokersTriggers extracts brokers and triggers data from the given config map.
+func (r *Reconciler) GetBrokersTriggers(logger *zap.Logger, brokersTriggersConfigMap *corev1.ConfigMap) (*coreconfig.Brokers, error) {
+
+	brokersTriggersRaw, hasData := brokersTriggersConfigMap.BinaryData[ConfigMapDataKey]
+	if !hasData || brokersTriggersRaw == nil {
+
+		logger.Debug(
+			fmt.Sprintf("config map has no %s key, so start from scratch", ConfigMapDataKey),
+		)
+
+		return &coreconfig.Brokers{}, nil
+	}
+
+	brokersTriggers := &coreconfig.Brokers{}
+	var err error
+
+	logger.Debug(
+		"unmarshalling configmap",
+		zap.String("format", r.Format),
+	)
+
+	// determine unmarshalling strategy
+	switch r.Format {
+	case PROTOBUF:
+		err = proto.Unmarshal(brokersTriggersRaw, brokersTriggers)
+	case JSON:
+		err = json.Unmarshal(brokersTriggersRaw, brokersTriggers)
+	}
+	if err != nil {
+
+		logger.Warn("failed to unmarshal config map", zap.Error(err))
+
+		// let the caller decide if it want to continue or fail on an error.
+		return &coreconfig.Brokers{}, fmt.Errorf("failed to unmarshal brokers and triggers: %w", err)
+	}
+
+	return brokersTriggers, nil
+}
+
+func (r *Reconciler) UpdateBrokersTriggersConfigMap(brokersTriggers *coreconfig.Brokers, configMap *corev1.ConfigMap) error {
+
+	var data []byte
+	var err error
+	switch r.Format {
+	case JSON:
+		data, err = json.Marshal(brokersTriggers)
+	case PROTOBUF:
+		data, err = proto.Marshal(brokersTriggers)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to marshal brokers and triggers: %w", err)
+	}
+
+	// Update config map data. TODO is it safe to update this config map? do we need to copy it?
+	configMap.BinaryData[ConfigMapDataKey] = data
+
+	_, err = r.KubeClient.CoreV1().ConfigMaps(configMap.Namespace).Update(configMap)
+	if err != nil {
+		return fmt.Errorf("failed to update config map %s/%s: %w", configMap.Namespace, configMap.Name, err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) UpdateDispatcherPodsAnnotation(logger *zap.Logger, volumeGeneration uint64) error {
+
+	namespace := r.SystemNamespace
+
+	labelSelector := labels.SelectorFromSet(map[string]string{"app": DispatcherLabel})
+	pods, errors := r.PodLister.Pods(namespace).List(labelSelector)
+	if errors != nil {
+		return fmt.Errorf("failed to list pods in namespace %s: %w", namespace, errors)
+	}
+
+	for _, p := range pods {
+
+		logger.Debug(
+			"update dispatcher pod annotation",
+			zap.String("pod", fmt.Sprintf("%s/%s", p.Namespace, p.Name)),
+			zap.Uint64("volumeGeneration", volumeGeneration),
+		)
+
+		// do not update cache copy
+		pod := p.DeepCopy()
+		p = nil
```

Comment:
My intention was to "disable" that variable so that misuses won't be possible.
I think it's not that difficult to use `p` instead of `pod` and then that bug is particularly difficult to find. If you have a better idea, let me know.

Reply:
why don't you just reuse p without creating a new variable?

Reply:
Make sense.